nucleus/modelci/data_transfer_objects/eval_function.py

```diff
@@ -0,0 +1,39 @@
+from typing import List, Optional
+
+from pydantic import BaseModel, validator
+
+from ..constants import ThresholdComparison
+
+
+class EvalFunctionCondition(BaseModel):
+    """
+    eval_function_id: ID of evaluation function
+    threshold_comparison: comparator for evaluation. i.e. threshold=0.5 and threshold_comparator > implies that a test only passes if score > 0.5.
+    threshold: numerical threshold that together with threshold comparison, defines success criteria for test evaluation.
+    """
+
+    # TODO: Having only eval_function_id hurts readability -> Add function name
```

Comment:
We initially went with eval function name but this quickly becomes a problem because of the concept of public eval functions. For each user, we enforce uniqueness of their evaluation function names, but we do not enforce global uniqueness of names, which means that public eval functions created by other users may have conflicting names.

We could address that by also enforcing uniqueness of names among a user's eval functions and public eval functions - I think that should be the way to go.

Reply:
Yeah, the name problem seems to keep popping up. We need to think about how we can make it as nice as we can for the user. These IDs are simply non-readable so if you're stepping through your code you have no idea what this eval function is. I don't know what the best way to go about this but we'll figure it out 🙂